### DISCUSSION TITLE
Socket Events Stop working

### DISCUSSION THREAD

**Discussion by SurbhiSardana** [socketio/socket.io#4939] [Q&A]
Hi

I am using socket.io client in a reactnative application, I am facing an issue that socket event stops working after some time if app is idle, not able to identify what actually happened, because it is not even disconnected, I didn't receive any disconnect event, so not able to reconnect.

I am using **socket.io-client version - 4.7.2** and  **react native version 0.70.6** . Here is my connection string 
const connectionConfig = {
  jsonp: false,
  reconnection: true,
  reconnectionDelay: 100,
  reconnectionAttempts: 100000,
  forceNew: true,
  upgrade: false,
  transports: ['websocket'],
  query: {
    source: 'mobile',
    platform: isIOS() ? IOS : ANDROID,
  },
};

**Comment by darrachequesne** [comment]
Hi!

Do you have any logs that could help us debugging your problem? Don't you get some "disconnect" or "connect_error" events?

Reference: https://socket.io/docs/v4/troubleshooting-connection-issues/

It may not be related, but is there any particular reason for `reconnectionAttempts` being not set to `Infinity`?

Reference: https://socket.io/docs/v4/client-options/#reconnectionattempts

**Reply by SurbhiSardana** [reply]
No, I didn't get any disconnect or connect error event
